glitch_app.py

```diff
@@ -109,86 +117,102 @@ def hash_file(filename: str) -> str:
     with open(filename, "rb") as f:
         return hashlib.md5(f.read()).hexdigest()
 
+
 def get_options(file_type, request) -> object:
-    opt = IMAGE_OPTIONS if file_type == 'image' else VIDEO_OPTIONS  
+    opt = IMAGE_OPTIONS if file_type == 'image' else VIDEO_OPTIONS
     return { key: opt[key]['type'](request.args.get(key)) for key in opt.keys() }
 
-@app.route("/glitch/<string:gid>", methods=["GET"])
-def glitch(gid):
-    filepath = signer.loads(gid)
-    file_type = get_file_type(filepath)
-    extension = file_extension(filepath)
-    file_hash = hash_file(filepath)
-    
-    options = get_options(file_type, request)
-    
-    print(f"Submitted options: {options}")
-
-    if not osp.exists(osp.join(STATIC_FOLDER, file_type, file_hash)):
-        os.makedirs(osp.join(STATIC_FOLDER, file_type, file_hash))
-        new_filepath = osp.join(
-            STATIC_FOLDER, file_type, file_hash, f"original.{extension}"
-        )
-        shutil.copy(filepath, new_filepath)
-        filepath = new_filepath
-        current_glitch_num = 0
-        other_glitches = []
-    else:
-        # glitch files are postfixed with -glitch-XX.ext
-        other_glitches = glob.glob(
-            osp.join(STATIC_FOLDER, file_type, file_hash, "*_glitch_*")
-        )
-        current_glitch_num = len(other_glitches)
 
-    glitched_fname = f"{file_hash}_glitch_{current_glitch_num}.{extension}"
-    current_glitch_num += 1
+@app.route("/glitch/<string:file_type>", methods=["GET", "POST"])
+def glitch(file_type):
+    options = IMAGE_OPTIONS if file_type == 'image' else VIDEO_OPTIONS
+    params  = { key: options[key]['type'](options[key]['default']) for key in options.keys() }
 
-    glitched_filepath = osp.join(STATIC_FOLDER, file_type, file_hash, glitched_fname)
+    if request.method == "POST":
+        reuse_last_file = False
 
-    if file_type == "image":
-        glitch_image(filepath, glitched_filepath, **options)
-    elif file_type == "video":
-        glitch_video(filepath, glitched_filepath, **options)
+        file = request.files.get("file", "")
+        if file == "" or file.filename == "":
```

Comment:
`file` is a python keyword, which is not recomended to shadow. use `filename` or something like this